src/Sarif/Baseline/ResultMatching/SarifLogResultMatcher.cs

```diff
@@ -175,8 +175,8 @@ private List<ExtractedResult> ExtractResultsFromRuns(IEnumerable<Run> sarifRuns,
                 {
                     foreach (Result result in run.Results)
                     {
-                        // Include all Results except Absent results in the baseline
-                        if (!(isBaselineRun && result.BaselineState == BaselineState.Absent))
+                        // Include all Results except Absent results
+                        if (result.BaselineState != BaselineState.Absent)
```

Comment:
This is a sensible way to go about things. 
A user complained about us ignoring Absent results in the baseline; he keeps them in his log and wants the baseline to have a rolling history (which I don't in SarifBaseliner).

If you're using Absent results in the current log to mean something, and want them to persist through to the output log, maybe you want them to come through but to be emitted in the state Absent as well.

This would mean that if you have a clear check that a condition is gone, you can emit an Absent result. Rather than hoping baselining will not match the old copy of the result to anything (and will therefore report the copy of the result from the Baseline as Absent) it would match the result with the older one, but see that the new one is Absent and write it out that way.

Reply:
The baseline is not an appropriate place for persisting complete history for many reasons you both understand. OTOH, it's quite reasonable for a current baseline to contain absent messages (though that have gone absent due to direct comparison to the prior baseline). What we should do is actively flush all absent messages from the baseline before beginning the diff.

---
In reply to: [397400299](https://github.com/microsoft/sarif-sdk/pull/1826#discussion_r397400299) [](ancestors = 397400299)